src/vm/amd64/unixasmhelpers.S

```diff
@@ -304,7 +232,7 @@ NESTED_ENTRY OnHijackInteriorPointerTripThread, _TEXT, NoHandler
 
     END_PROLOGUE
 
-    call                C_FUNC(OnHijackInteriorPointerWorker)
+    call                C_FUNC(OnHijackWorker)
 
     free_stack          0x8
     pop_register        rax
```

Comment:
Ah, I am sorry, the diff has hidden the RDX manipulation and I have missed that.
As for the possibility of GC reference and floating point in the return value, yes, it is possible. It is valid to return a struct with one reference and one floating point in registers.